pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -132,11 +160,45 @@ public synchronized void addConsumer(Consumer consumer) throws BrokerServiceExce
             throw new ConsumerBusyException("Subscription reached max consumers limit");
         }
 
+        deadLetterTopicProducer = newDeadLetterProducer();
+
         consumerList.add(consumer);
         consumerList.sort((c1, c2) -> c1.getPriorityLevel() - c2.getPriorityLevel());
         consumerSet.add(consumer);
     }
 
+    private ProducerImpl<byte[]> newDeadLetterProducer() throws BrokerServiceException {
+        if (maxRedeliveryCount > 0 && deadLetterTopicProducer == null) {
+            try {
+                if (maxRedeliveryCount > 0 && StringUtils.isBlank(deadLetterTopic)) {
+                    deadLetterTopic = String.format("%s-%s-DLQ", topic.getName(), Codec.decode(cursor.getName()));
+                }
+                return (ProducerImpl<byte[]>) topic.getBrokerService().pulsar().getClient().newProducer(Schema.BYTES)
+                        .topic(deadLetterTopic)
+                        .enableBatching(false)
```

Comment:
any thoughts about why you need to disable batching?

Reply:
No, i will enable batching